src/crossChain/AxelarProvider.ts

```diff
@@ -69,11 +81,47 @@ export class AxelarProvider implements CrossChainProvider {
 
     // @ts-ignore
     // tslint:disable-next-line
-    const txResponse = (await this.squid.executeRoute({
-      route: route.route,
-      signer
-    })) as ethers.providers.TransactionResponse
+    let txResponse: ethers.providers.TransactionResponse | null = (await this.squid.executeRoute(
+      {
+        route: route.route,
+        signer
+      }
+    )) as ethers.providers.TransactionResponse
 
+    const originChainTxHash = txResponse.hash
+
+    // if it's trigger from a different network than MATIC, we need to fetch the MATIC tx receipt
+    if (route.route.params.fromChain !== ChainId.MATIC_MAINNET.toString()) {
+      let status: StatusResponse | undefined
+      try {
+        status = await this.squid.getStatus({
+          transactionId: txResponse.hash,
+          integratorId: INTEGRATOR_ID,
+          requestId: route.requestId
+        })
+      } catch (error) {
+        console.error('error: ', error)
+      }
+      txResponse = null
+      const polygonRPC = getRpcUrls(ProviderType.NETWORK)[ChainId.MATIC_MAINNET]
+      const polygonProvider = new ethers.providers.JsonRpcProvider(polygonRPC)
```

Comment:
Should we consider waiting for the tx receipt from the ethereum network as well in order to consider the minting of names, L1 wearables and lands?
What if we get from the route the chain id of the asset and use it here to build the provider to wait the transaction for?